### ISSUE TITLE
Hangs on "connecting"

### ISSUE THREAD

**Issue by WhiteGrizly** [Freeyourgadget/Gadgetbridge#1238]
- [*] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [*] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.

29.1 Gatgetbridge hangs on "connecting" ("Соединение" на русском) 
on 28.1 all was good.

*Please specify model and firmware version if possible*
Amazfit bip 1.0.1.00

#### Your android version is:
miui global 9.2.4.0 android 7.1.2.N2G47H

#### Your Gadgetbridge version is:

0.29.1

**Comment by ashimokawa** [comment]
Can you try unpairing in android settings, toggle BT off/on, and then press the + button in Gadgetbridge and pair again?

**Comment by WhiteGrizly** [comment]
I found the time and the Internet did as you wrote. The problem continues. GB writes "connection", and on bip writes that it is connected. This happens the second morning. Day and night was not.

**Comment by ahmadalsaady** [comment]
I am having the same problem connecting to my amazfit bip its not connecting automatically and it is stuck at connecting even it is connected

**Comment by MagicFab** [comment]
This happened to me recently after flashing / upgrading new ROMs.

If you haven't yet, I'd suggest resetting network settings ([here is an example procedure for Android 7/Nougat](https://help.republicwireless.com/hc/en-us/articles/115012387128-How-to-Reset-Network-Settings-on-Samsung-Galaxy-Phones-on-Android-7-0-or-Higher)). Please note [other settings than Bluetooth will also be deleted / reset](https://support.sonymobile.com/global-en/xperiam4aqua/faq/apps-&-settings/80193074012a2df1c0151cca06151007319/).

Also perhaps consider upgrading, hopefully your ROM is available for more recent Android versions.

**Comment by WhiteGrizly** [comment]
I don't update the firmware on the devices if they are working properly. In addition, I have a large database of wi-fi networks and passwords around the world, I do not want to lose it. Didn't change anything and wasn't going to. Just the GB version. Destroying data, updating versions and other serious problems only because of one program is not good idea, as for me...

**Comment by joserebelo** [comment]
If I understood you correctly, does the watch say that it is connected? What does the Gadgetbridge notification say? I used to have a problem where the connection was successful, but the device list stayed connecting.

Also, a [log file](https://github.com/Freeyourgadget/Gadgetbridge/wiki/Log-Files) might be useful.

**Comment by WhiteGrizly** [comment]
>the watch say that it is connected?
Yes
>Gadgetbridge notification say?
GB show "connecting" and thats all. I can`t view fit. info, notifications absent.

**Comment by MagicFab** [comment]
Can you reinstall 28.1 and confirm it works as expected? Then perhaps logs could be compared.
Upgrading is not a "serious problem", it's good practice specially for security reasons. Your network settings can be backed up to test this, perhaps using a new profile could help isolate the issue.

**Comment by WhiteGrizly** [comment]
At the top I already wrote: "on 28.1 everything was good".

Upgrade is a serious problem, as it is known that the new version even more unnecessary applications, besides have to re-clean out all of the phone and memory. The best security is network protection with a third-party firewall. There are 2 security problems on smartphones, 1-if it falls into the wrong hands (then do not help the update) 2 - sending confidential data to third-party servers, but with each new version Google only adds such features to their applications, Chinese manufacturers miui not lag behind. But this is a departure from the topic. I will not update the system. If it's for GB, I'll stop using it.

**Comment by MagicFab** [comment]
After some research I understand now MIUI are stock ROMs that come with Xiaomi phones, so I assume not having root you can't backup and restore all apps and their settings conveniently, so yes, it's a huge hassle to upgrade, specially when a full reset is required.

I am using an Amazfit Bit with both Lineage OS 14 and 15 (so, a bit more recent Android versions) and F/W is 1.1.1.00, without issues before or after 0.28.1 / 0.29.1.

My suggestion to reinstall 0.28.1 was to get a log of the device working and connecting successfully, then upgrade, and then get a new log and submit both here for analysis.

**Comment by WhiteGrizly** [comment]
You, maybe, not have a problem because we have different terms of use. I am now in the forest and occasionally the watch loses contact with the phone, so the problem is detected more often. The logs and reinstalling I now have to do there is no way, and I have Internet through VPN here.

**Comment by WhiteGrizly** [comment]
Problem is NOT solved. 
![img_20180921_210645](https://user-images.githubusercontent.com/41453327/45899981-04c64d80-bde7-11e8-91da-a3902deca1f6.jpg)

Bip show that it is "connected", GB hangs on "connecting".

**Comment by cpfeiffer** [comment]
Not solved in which version? 0.30?

**Comment by WhiteGrizly** [comment]
Sorry, yes, 0.30

**Comment by TrMaXXX** [comment]
I have the same problem:  
Amazfit Bip v1.1.2.05
GadgetBridge 0.30

**Comment by TrMaXXX** [comment]
> Would it be possible for one of you to provide a log file created with 0.30?
I did unpairing in MiFit app and then pairing directly with GadgetBridge.
It solved the problem for me.

**Comment by WhiteGrizly** [comment]
For me, this solution (unpairing with mi fit) is impossible, in GB there are no reminders and the ability to manage alarms as in mi fit. Log is turned on, now the worst-only loses connection and no longer tries to connect, the case as in the photo was not yet.

**Comment by TrMaXXX** [comment]
> For me, this solution (unpairing with mi fit) is impossible, in GB there are no reminders and the ability to manage alarms as in mi fit. Log is turned on, now the worst-only loses connection and no longer tries to connect, the case as in the photo was not yet.

It seems to me that the problem depends on whether "Fitness Notify for Amazfit" is installed and paired with watch, Do you use this App?

**Comment by WhiteGrizly** [comment]
No. Mifit was used to bind bip, to include visibility, the watch is connected with GB, mifit turned off the visibility and unloaded. That's all. I did not install any other programs.

**Comment by WhiteGrizly** [comment]
The second problem is also confirmed. When working with GB, the rate of battery loss increases by more than 2 times. Instead of 2% 4-7% with the same use. Tested 2 weeks with mifit (2%) and now since the release and installation of version 0.30 to today 4-7% .

**Comment by WhiteGrizly** [comment]
Did not wait for the problem, turned off the logs, the problem immediately appeared. Included log back in before prepollical watch. Need such logs? But the hang was clearly much less.

**Comment by WhiteGrizly** [comment]
[gadgetbridge.log](https://github.com/Freeyourgadget/Gadgetbridge/files/2454365/gadgetbridge.log)

This log was turned on at the time of the glitch that had already occurred, then bluetooth was turned off and on and the connection was established. Unfortunately, I can't leave a permanent record on the phone's built-in memory for weeks, it's too wasteful.

**Comment by free5lot** [comment]
> Just asking, @free5lot do you have the official app installed as well

@TaaviE thanks for fast response. I had official app on different phone and it once was connected to this device, but actually no, I'm not using official app for months on any phone now. 
Today I made a reset on my Amazfit Cor (using band's gui menu) to drop any link to the original app, and connected it only to Gadgetbridge for now. I'll let you know in this issue if I have this issue again.

I'm not ready to share detailed information about my phone, but maybe this will be useful - I'm using Gadgetbridge from F-Droid on LineageOS (Android 8) and Gadgetbridge was updated recently to 0.32.1 and the issue still exists.

Gadgetbridge team should really consider to add timeout option for this "Connecting" state. Because if "Connecting" stays for some time, like a minute or more, it means definitely something is wrong and maybe reconnect attempt is required (if auto-reconnects are turned on in settings).

**Comment by dmig** [comment]
Issue #1452 is a duplicate of this.

My case: Mi Band 2 doesn't reconnect automatically **before Mi Fit is connected**. The latter is the key.

**Comment by dmig** [comment]
Here is my log:
[gadgetbridge.log](https://github.com/Freeyourgadget/Gadgetbridge/files/2990816/gadgetbridge.log)

**Comment by free5lot** [comment]
> Today I made a reset on my Amazfit Cor (using band's gui menu) to drop any link to the original app, and connected it only to Gadgetbridge for now. I'll let you know in this issue if I have this issue again.

The issue is still present and very annoying - you are always not sure if the device is really connected!

>Gadgetbridge team should really consider to add timeout option for this "Connecting" state. Because if "Connecting" stays for some time, like a minute or more, it means definitely something is wrong and maybe reconnect attempt is required (if auto-reconnects are turned on in settings).

It's definitely possible to fix this issue, because the Gadgetbidge obviously knows that it's "trying" to connect for ridiculous period of time.
Please implement timeouts for connecting state and add more flexible reconnect logic (configurable using settings).

**Comment by dmig** [comment]
Version 0.33 still has this issue. But I found a workaround: pairing band with phone fixes reconnection, probably by delegating it to system bluetooth stack.

**Comment by free5lot** [comment]
> Version 0.33 still has this issue.
Yes, I also still have this issue, like almost everyday.

> But I found a workaround: pairing band with phone fixes reconnection, probably by delegating it to system bluetooth stack.
I will take a look, thank you for information.

Actually this issue is the most annoying thing in the Gadgetbridge and I would give it top1 priority, because it decreases the reliability of the daily use to zero. You're never 100% sure if the device is really connected or not without checking the phone. :(

Also this issue can be easily fixed at least by adding timeout for "Reconnecting" process, because now the situation is ridiculous - the Gadgetbridge waits for single reconnection indefinitely instead of dropping connection and trying to connect again and again.

**Comment by mjanssens** [comment]
What I did:
DB management, empty DB
After that tried to pair the mi band 3 in Gadgetbirdge, but didn't succeed.
Saw that I had a bluetooth pairing in android settings with the mi band 3, so deleted that.
Paired again in Gadgedbridge, now succesful.
I have had one  full day now without disconnected mi band 3.
Current FW 2.3.0.28 HW V0.18.4.1
Gadgetbridge updated half a day after that to 0.33.1 (don't know if this also has effect)

**Comment by IzzySoft** [comment]
@WhiteGrizly can we consider this issue solved? Did @mjanssens steps help you?

**Comment by free5lot** [comment]
> can we consider this issue solved?
Not fixed. Still have this issue (version 0.35.2).
How was it fixed, with what code/commit exactly? Because I'm not seeing anything about fixing this problem in changelog. It's a huge unsolved issue to me.

**Comment by IzzySoft** [comment]
@free5lot see the comment above my previous one. I didn't claim a commit here having fixed it (could well have been a firmware update, or a broken setting). So have you tried the steps mentioned by mjanssens?

**Comment by free5lot** [comment]
@IzzySoft yes, I have this issue on two different phones, tried to uninstall GadgetBridge, cleaning cache, tried reconnecting, changing mac address, updating band's firmware using original Xiaomi proprietary application and so on. The problem stays.

One more point - why this "Connecting forever" state is considered to be OK by GadgetBridge at all? Obviously short timeout (less than 1 minute) is required anyway, now it's kind of a bug - app says it's "Connecting" but actually it's in some incorrect state for unlimited amount of time.

Please look at my previous comments in this ticket, I provided logical ways to fix this issue by adding timeouts. Is it fine?

**Comment by IzzySoft** [comment]
Thanks for your feedback, @free5lot! I'm not a GB dev, I just help out with issue maintenance so the devs can focus on development, so I cannot answer your questions. So I was checking if this issue is still relevant (unfortunately, Carsten is currently unavailable, so this might be why there was no further response from the devs' side).

Now with the facts established, maybe @ashimokawa can take a look and pick up from where Carsten left.

**Comment by free5lot** [comment]
Thanks @IzzySoft, I'm looking forward to see this being fixed, the main problem for me with GB for now.

**Comment by dmig-alarstudios** [comment]
I'm keeping an eye on this issue for a long time since I'm suffering from this bug also. One of the triggers is the activity fetching -- often leaves app in unconscious state.
